jax_triton/triton_lib.py

```diff
@@ -226,8 +260,85 @@ def compile_ttir_to_ptx_inplace(
   cluster_dims = metadata["cluster_dims"]
   ttgir = str(ttgir) if _JAX_TRITON_DUMP_DIR else None
   llir = str(llir) if _JAX_TRITON_DUMP_DIR else None
-  return PtxCompilationResult(
-      ptx=ptx,
+  return CompilationResult(
+      ptx_or_hsaco=ptx,
+      name=name,
+      shared_mem_bytes=shared_mem_bytes,
+      cluster_dims=cluster_dims,
+      ttgir=ttgir,
+      llir=llir,
+  )
+
+def compile_ttir_to_hsaco_inplace(
+    ttir,
+    hip_backend: hb.HIPBackend,
+    hip_options: hb.HIPOptions,
+    compute_capability,
+) -> CompilationResult:
+  if hip_options.debug:
+    print(ttir)
+  if isinstance(ttir, ir.Module):
```

Comment:
I don't think we need this branch any more tbh. It was added back when we were migrating Pallas off Triton IR builders.

Reply:
@superbobry To clarify you mean no need to check for this "if isinstance(ttir, ir.Module):" Or you mean, remove the if statement along with statements under it?

Reply:
Yeah, but let me do this in a follow up, actually :)